src/seekevaluate.h

```diff
@@ -37,16 +37,55 @@
 
 namespace Sleipnir {
 
+    template <typename K, typename V>
+    struct PairedResult {
+        K key;
+        V val;
+
+        PairedResult() {}
+
+        PairedResult(K _key, V _val) : key(_key), val(_val) {}
+
+        bool operator<(const PairedResult &item) const {
+            if (val == item.val && key == item.key) {
+                return false;
```

Comment:
I remember you mentioned it why is it done this way but nevertheless it strange when it returns false not true when key and value is the same...

Reply:
This less-than operator is generally used for sorting vectors. A return value of true implies that they are out of order. But when the two are equal we need to return false or we can get into an infinite loop where it keeps swapping the two equal values.